projects/types.nix

```diff
@@ -253,25 +253,49 @@ let
         }
       );
 
-    demo = types.submodule {
-      options = {
-        inherit (types'.example.getSubOptions { })
-          module
-          tests
-          description
-          links
-          ;
-        problem = mkOption {
-          type = types.nullOr types'.problem;
-          default = null;
-          example = {
-            problem.broken = {
-              reason = "Does not work as intended. Needs fixing.";
+    demo = types.submodule (
+      { name, config, ... }:
+      {
+        options = {
+          inherit (types'.example.getSubOptions { })
+            module
+            tests
+            description
+            links
+            ;
+          # TODO: why doesn't this work with an enum?
```

Comment:
That's very weird. What's the error? (likely something to do with lazyness??)

Reply:
`null` would produce a type error though, no?

Reply:
Yeah, not sure. I'm just speculating, because the type can't be anything else since we're using an attrTag.